### PR TITLE
Allow unnamed functions in wasm-dis#readFunctions, resolves #107

### LINKED ISSUE #107: wasm-dis: functions need not be named.
`readFunctions()` asserts that functions are named, but only imports and exports are required to be named.

### PR DESCRIPTION
**Pull request by bsklaroff**
Now, unnamed functions in binaries are allowed and automatically assigned a numeric name.

Tested manually by modifying wasm-as not to write function names, then disassembling those binaries with wasm-dis. Not sure how to automatically test this within the current framework since wasm-as writes names for all functions by default.

### CODE REVIEW

File: src/wasm-binary.h [hunk 1]
```diff
@@ -1119,8 +1121,8 @@ class WasmBinaryBuilder {
         e->value = name;
```
Comment (kripken) [inline src/wasm-binary.h]: if name is not set, then we are setting that null string here, which is bad.

we would have to autogenerate a name earlier, in parallel to the `getString`.

Comment (kripken) [inline src/wasm-binary.h]: alternatively, we could create the export later down, after we have called `addImport/addFunction`, which would have set a numeric name.

**Comment by kripken** [comment]
Good question about testing. It seems like we have to either enable `wasm-as` to emit unnamed functions, or to have tests with the actual binary format. But as the binary format is still very much undefined and changing, the second option sounds bad.

**Comment by bsklaroff** [comment]
this is outdated, closing

**Pull request closed without merging**